### ISSUE TITLE
Sigmoid overflows on large finite negative logits

### ISSUE THREAD

**Issue by vijayksingh** [unlocalhosted/browsergrad#8]
## What happens

Published BrowserGrad 0.8.2 and the current 0.9.0 source implement sigmoid as `1 / (1 + exp(-x))`. A large finite negative input therefore overflows the exponential. Forward happens to approach zero, but emits a runtime warning, and the decomposed backward can be poisoned by the intermediate infinity.

## Minimal reproduction

```python
import browsergrad_jit as bg

x = bg.tensor([-1000.0, 0.0, 1000.0], dtype="float64", requires_grad=True)
y = bg.sigmoid(x)
y.sum().backward()

print(y.tolist())
print(x.grad.tolist())
```

Expected:

```text
[0.0, 0.5, 1.0]
[0.0, 0.25, 0.0]
```

Expected behavior also includes no overflow warning for finite inputs.

## Suggested direction

Use the sign-split stable formula:

- `x >= 0`: `1 / (1 + exp(-x))`
- `x < 0`: `exp(x) / (1 + exp(x))`

For autograd, the closed-form local derivative `y * (1 - y)` avoids differentiating through an overflowing decomposition. Tests should cover float32/float64 extreme finite logits and gradients.

**Comment by vijayksingh** [comment]
> *This was generated by AI during triage.*

## Agent Brief

**Category:** bug
**Summary:** Make sigmoid forward and backward numerically stable for large finite logits.

**Current behavior:**
Sigmoid is represented as `1 / (1 + exp(-x))`. In a fresh Pyodide check, inputs `[-1000, 1000]` realize to `[0, 1]`, but backward produces `[NaN, 0]`. The negative branch overflows internally even though the mathematically correct output and gradient are finite.

**Desired behavior:**
Sigmoid should use a sign-stable semantic formulation for finite inputs and its derivative should be computed from a stable sigmoid result. The behavior must compose through the public function, tensor method/torch alias where exposed, and lazy backward without weakening generic `exp` diagnostics.

**Key interfaces:**
- `sigmoid()` — stable forward behavior for positive and negative finite inputs.
- Sigmoid VJP/autograd rule — return finite `y * (1-y)` gradients at saturation.
- Lazy realization and dtype handling — preserve supported float32/float64 behavior.

**Acceptance criteria:**
- [ ] Float32 and float64 sigmoid at `-1000` and `1000` returns finite values equivalent to `0` and `1` without overflow/invalid warnings.
- [ ] Backward for those inputs returns finite zero-limit gradients and never NaN.
- [ ] Moderate positive, negative, and zero inputs match a stable NumPy reference for forward and backward.
- [ ] Public aliases and module consumers use the same stable semantics.
- [ ] Existing fusion/lazy execution tests remain passing.

**Out of scope:**
- Changing the semantics of generic exponentiation.
- Adding unrelated activation functions.

**Comment by vijayksingh** [comment]
> *This was generated by AI during triage.*

## Verified outcome

**State:** wontfix — already implemented

Implemented in https://github.com/unlocalhosted/browsergrad/commit/4d7181a9e14ba8c69e8714277da4172b361afcf4: sigmoid uses an overflow-safe typed lazy formulation. Float32/float64 logits at -1000, 0, and 1000 produce finite forward values and gradients without overflow warnings, including derivative 0.25 at zero. Fresh verification passed JIT build, both typechecks, lint, 27 unit tests, 345 integration tests, architecture checks, packed-release tests, and the required JIT semantic-permute WebGPU gate.

Closing as completed and replacing the ready state with `wontfix` per the repository triage state machine.
